File: solutions/Golang/clone_graph_bfs.go
package main

type Node struct {
    Val       int
    Neighbors []*Node
}

func cloneGraphBFS(node *Node) *Node {
    if node == nil {
        return nil
    }
    
    visited := make(map[int]*Node)
    queue := []*Node{node}
    visited[node.Val] = &Node{Val: node.Val}
    
    for len(queue) > 0 {
        curr := queue[0]
        queue = queue[1:]
        
        for _, neighbor := range curr.Neighbors {
            if _, ok := visited[neighbor.Val]; !ok {
                visited[neighbor.Val] = &Node{Val: neighbor.Val}
                queue = append(queue, neighbor)
            }
            visited[curr.Val].Neighbors = append(visited[curr.Val].Neighbors, visited[neighbor.Val])
        }
    }
    
    return visited[node.Val]
}

File: solutions/JavaScript/clone_graph_bfs.js
class Node {
    constructor(val = 0, neighbors = []) {
        this.val = val;
        this.neighbors = neighbors;
    }
}

function cloneGraphBFS(node) {
    if (!node) return null;
    
    const visited = new Map();
    const queue = [node];
    visited.set(node.val, new Node(node.val));
    
    while (queue.length > 0) {
        const curr = queue.shift();
        
        for (const neighbor of curr.neighbors) {
            if (!visited.has(neighbor.val)) {
                visited.set(neighbor.val, new Node(neighbor.val));
                queue.push(neighbor);
            }
            visited.get(curr.val).neighbors.push(visited.get(neighbor.val));
        }
    }
    
    return visited.get(node.val);
}
